scripts/cmake/vcpkg_fixup_pkgconfig.cmake

```
@@ -45,149 +45,165 @@ Still work in progress. If there are more cases which can be handled here feel f
+            if("${config}" STREQUAL "DEBUG")
+                # prefix points at the debug subfolder
+                string(REPLACE [[${prefix}/debug]] [[${prefix}]] contents "${contents}")
+                string(REPLACE [[${prefix}/include]] [[${prefix}/../include]] contents "${contents}")
+                string(REPLACE [[${prefix}/share]] [[${prefix}/../share]] contents "${contents}")
             endif()
-            string(REGEX REPLACE " -L(\\\${[^}]*}[^ \n\t]*)" " -L\"\\1\"" _contents "${_contents}")
-            string(REGEX REPLACE " -I(\\\${[^}]*}[^ \n\t]*)" " -I\"\\1\"" _contents "${_contents}")
-            string(REGEX REPLACE " -l(\\\${[^}]*}[^ \n\t]*)" " -l\"\\1\"" _contents "${_contents}")
+            # quote -L, -I, and -l paths starting with `${blah}`
+            string(REGEX REPLACE " -([LIl])(\\\${[^}]*}[^ \n\t]*)" [[ -\1"\2"]] contents "${contents}")
             # This section fuses XYZ.private and XYZ according to VCPKG_LIBRARY_LINKAGE
             #
             # Pkgconfig searches Requires.private transitively for Cflags in the dynamic case,
             # which prevents us from removing it.
             #
             # Once this transformation is complete, users of vcpkg should never need to pass
             # --static.
-            if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-                # Libs comes before Libs.private
-                string(REGEX REPLACE "(^|\n)(Libs: *[^\n]*)(.*)\nLibs.private:( *[^\n]*)" "\\1\\2\\4\\3" _contents "${_contents}")
-                # Libs.private comes before Libs
-                string(REGEX REPLACE "(^|\n)Libs.private:( *[^\n]*)(.*\nLibs: *[^\n]*)" "\\3\\2" _contents "${_contents}")
-                # Only Libs.private
-                string(REGEX REPLACE "(^|\n)Libs.private: *" "\\1Libs: " _contents "${_contents}")
-                # Requires comes before Requires.private
-                string(REGEX REPLACE "(^|\n)(Requires: *[^\n]*)(.*)\nRequires.private:( *[^\n]*)" "\\1\\2\\4\\3" _contents "${_contents}")
-                # Requires.private comes before Requires
-                string(REGEX REPLACE "(^|\n)Requires.private:( *[^\n]*)(.*\nRequires: *[^\n]*)" "\\3\\2" _contents "${_contents}")
-                # Only Requires.private
-                string(REGEX REPLACE "(^|\n)Requires.private: *" "\\1Requires: " _contents "${_contents}")
+            if("${VCPKG_LIBRARY_LINKAGE}" STREQUAL "static")
+                # how this works:
+                # we want to transform:
+                #   Libs: $1
+                #   Libs.private: $2
+                # into
+                #    Libs: $1 $2
+                # and the same thing for Requires and Requires.private
+
+                foreach(item IN ITEMS "Libs" "Requires")
+                    set(line "")
+                    if("${contents}" MATCHES "(^|\n)${item}: *([^\n]*)")
+                        string(APPEND line " ${CMAKE_MATCH_1}")
```

```suggestion
                        string(APPEND line " ${CMAKE_MATCH_1}")
```
Doesn't this just match the `(^|\n)` group?

---

yes